pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -339,9 +340,57 @@ func startModuleConfiguration(description string, configureModules func() error)
 }
 
 // Ceph docs about the prometheus module: http://docs.ceph.com/docs/master/mgr/prometheus/
-func (c *Cluster) enablePrometheusModule() error {
-	if err := cephclient.MgrEnableModule(c.context, c.clusterInfo, PrometheusModuleName, true); err != nil {
-		return errors.Wrap(err, "failed to enable mgr prometheus module")
+func (c *Cluster) configurePrometheusModule() error {
+	if !c.spec.Prometheus.Disabled {
+		if err := cephclient.MgrEnableModule(c.context, c.clusterInfo, PrometheusModuleName, true); err != nil {
+			return errors.Wrap(err, "failed to enable mgr prometheus module")
+		}
+	} else {
+		if err := cephclient.MgrDisableModule(c.context, c.clusterInfo, PrometheusModuleName); err != nil {
+			logger.Errorf("failed to disable mgr prometheus module. %v", err)
+		}
+		return nil
+	}
+
+	var (
+		err        error
+		hasChanged = false
+		changed    = false
+		daemonID   = "mgr"
+	)
+	monStore := config.GetMonStore(c.context, c.clusterInfo)
+	// port
+	if c.spec.Prometheus.Port != 0 {
+		port := strconv.Itoa(c.spec.Prometheus.Port)
+		hasChanged, err = monStore.SetIfChanged(daemonID, "mgr/prometheus/server_port", port)
+		if err != nil {
+			return err
+		}
+	}
+	// scrape interval
+	if c.spec.Prometheus.Interval != nil {
+		interval := c.spec.Prometheus.Interval.Duration.Seconds()
+		changed, err = monStore.SetIfChanged(daemonID, "mgr/prometheus/scrape_interval", fmt.Sprintf("%v", interval))
+		if err != nil {
+			return err
+		}
+		hasChanged = hasChanged || changed
+	}
+
+	if hasChanged {
+		logger.Info("prometheus config has changed. restarting the prometheus module")
+		return c.restartMgrModule(PrometheusModuleName)
+	}
+	return nil
+}
+
+func (c *Cluster) restartMgrModule(name string) error {
```

Comment:
There is a `restartDashboard()` that does something similar, we can remove it and have it call this method since it's more general

Reply:
done